src/services/dealer-price/index.ts

```diff
@@ -0,0 +1,116 @@
+import util from "util"
+
+import { credentials } from "@grpc/grpc-js"
+
+import { UnknownDealerPriceServiceError } from "@domain/dealer-price"
+
+import { baseLogger } from "../logger"
+
+import { PriceServiceClient } from "./proto/services/price/v1/price_service_grpc_pb"
+import {
+  GetExchangeRateForImmediateUsdBuyRequest,
+  GetExchangeRateForImmediateUsdBuyResponse,
+  GetExchangeRateForImmediateUsdSellRequest,
+  GetExchangeRateForImmediateUsdSellResponse,
+  GetExchangeRateForFutureUsdBuyRequest,
+  GetExchangeRateForFutureUsdBuyResponse,
+  GetExchangeRateForFutureUsdSellRequest,
+  GetExchangeRateForFutureUsdSellResponse,
+} from "./proto/services/price/v1/price_service_pb"
+
+const serverPort = process.env.PRICE_SERVER_PORT ?? "50055"
+const client = new PriceServiceClient(
+  `localhost:${serverPort}`,
+  credentials.createInsecure(),
+)
+
+const clientGetExchangeRateForImmediateUsdBuy = util.promisify(
+  client.getExchangeRateForImmediateUsdBuy.bind(client),
+)
+const clientGetExchangeRateForImmediateUsdSell = util.promisify(
+  client.getExchangeRateForImmediateUsdSell.bind(client),
+)
+const clientGetExchangeRateForFutureUsdBuy = util.promisify(
```

Comment:
maybe `clientGetQuoteRateForFutureUsdBuy` is a good name?

if you don't like the use of option, I think the wording `quote` convey the optionality?